tls/s2n_handshake_io.c

```diff
@@ -510,16 +515,21 @@ int s2n_generate_new_client_session_id(struct s2n_connection *conn)
 /* Lets the server flag whether a HelloRetryRequest is needed while processing extensions */
 int s2n_set_hello_retry_required(struct s2n_connection *conn)
```

Comment:
Is this method tested? I don't see any test changes to reflect the new failure cases

Reply:
This method was added with the server HRR tests. Seems like there were no tests written to verify even the old state. I can add it.